### PR TITLE
Feature/subway classes

### PR DESCRIPTION
**Pull request by kamatozzzik**

### CODE REVIEW

File: src/css/style.css [hunk 1]
```diff
@@ -0,0 +1,82 @@
+p {
+	margin: 0;
```
Comment (AndreiShenets) [inline src/css/style.css]: Tabs here...

File: src/js/Route.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export class Route {
+    constructor(stations) {
+        this.stations = [];
+        this._setRoute(stations);
+        this.length = this.stations.length;
+    }
+
+    _setRoute(stations) {
```
Comment (AndreiShenets) [inline src/js/Route.js]: Why do you need this method? why just don't assign stations in constructor?

File: src/js/Station.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Station {
+    constructor(name) {
+        this.name = name;
+        this.connections = [];
```
Comment (AndreiShenets) [inline src/js/Station.js]: Why is this here?

File: src/js/Station.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+        this.siblings = [];
```
Comment (AndreiShenets) [inline src/js/Station.js:4]: Why is this here?

File: src/js/Station.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    }
+
+    isLine(line) {
```
Comment (AndreiShenets) [inline src/js/Station.js]: How station can be a line? Why is this here?

File: src/js/Subway.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export class Subway {
```
Comment (AndreiShenets) [inline src/js/Subway.js:1]: How are you going to modify this? You don't pass you lines into constructor and do not have methods for modification...

File: src/js/SubwayLine.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+export class SubwayLine {
+    constructor(name, isCicle, stations) {
```
Comment (AndreiShenets) [inline src/js/SubwayLine.js]: Why do you need isCircle parameter if you can determine it from stations?

File: src/js/SubwayLine.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+        this.name = name;
+        this.isCicle = isCicle;
+        this.stations = stations;
+        this.connections = [];
+    }
+
+    hasStation(stationName) {
+        return this.stations.some(station => {
+            return stationName === station.name ? true : false;
```
Comment (AndreiShenets) [inline src/js/SubwayLine.js]: Is it case sensitive or case insensitive comparison?

File: src/js/startApp.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+import { SubwayLine } from './SubwayLine';
+import { Subway } from './Subway';
+import { Route } from './Route';
+
+let fileReader = new FileReader();
+let textInput = document.querySelector('#file-input');
+
+textInput.addEventListener('change', function(e) {
+    file = e.target.files[0];
+    fileReader.readAsBinaryString(file);
```
Comment (AndreiShenets) [inline src/js/startApp.js:12]: Result of reading is not stored or used

**Review by AndreiShenets — CHANGES_REQUESTED** [review]

File: src/js/Station.js (hunk 4 above)
Comment (kamatozzzik) [reply src/js/Station.js:4]: It's for adding of nearest stations for search of routes

File: src/js/Station.js (hunk 3 above)
Comment (kamatozzzik) [reply src/js/Station.js]: Connections are for a detection of line one belongs

File: src/js/Station.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
+class Station {
```
Comment (AndreiShenets) [inline src/js/Station.js]: No export here

File: src/js/SubwayLine.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+export class SubwayLine {
+    constructor(name, stations) {
+        this.name = name;
+        this.stations = stations;
+        this.connections = [];
```
Comment (AndreiShenets) [inline src/js/SubwayLine.js]: ????

File: src/js/SubwayLine.js [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+    }
+
+    hasStation(stationName) {
+        return this.stations.some(station => {
+            let currentName = station.name.toLowerCase();
+            let requestName = stationName.toLowerCase();
+            
+            return currentName === requestName ? true : false;
```
Comment (AndreiShenets) [inline src/js/SubwayLine.js]: ???

**Review by AndreiShenets — CHANGES_REQUESTED** [review]

**Comment by kamatozzzik** [comment]
All fixed

**Review by AndreiShenets — APPROVED** [review]
